### DISCUSSION TITLE
执行 'osqp' 类析构函数时，捕获到以下错误: 未定义与 'char' 类型的输入参数相对应的函数 'osqp_mex'。

### DISCUSSION THREAD

**Discussion by styw1234** [yalmip/YALMIP#1073] [Problems using YALMIP] [answered]
When I use yalmip, it comes out these problems:

警告: 执行 'osqp' 类析构函数时，捕获到以下错误:
未定义与 'char' 类型的输入参数相对应的函数 'osqp_mex'。

出错 osqp/delete (第 40 行)
            osqp_mex('delete', this.objectHandle);

出错 osqp (第 34 行)
            this.objectHandle = osqp_mex('new', varargin{:});

出错 sdpsettings>setup_osqp_options (第 1222 行)
    s = osqp;

出错 sdpsettings (第 244 行)
    options.osqp = setup_osqp_options;

出错 test07 (第 15 行)
options = sdpsettings('solver','cplex'); 
> 位置：osqp (第 34 行)
位置: sdpsettings>setup_osqp_options (第 1222 行)
位置: sdpsettings (第 244 行)
位置: test07 (第 15 行)

**Comment by johanlofberg** [comment, marked as answer]
I don't read chinese, but you have OSPQ installed, and MATLAB is telling you that you OSQP does not work

**Comment by straysheep33** [comment]
也许有帮助https://blog.csdn.net/Utakuta/article/details/113180835
但我自己试了之后警告还是在，但是不影响问题的求解

**Comment by Qinxinke** [comment]
我也遇到这个问题，看网上大概三种说法
第一种是不要将matpowerX.X文件夹放入matlab安装目录。我试着重新下了改目录还是有错
第二种是说没有装C编译器，另一个老哥说的那个网址就是解决这个的，但很可惜对我来说还是没用
最后一种是yalmip和matpower里的osqp函数复用冲突，具体方法是出错那个信息点击下，会进入到报错的那个文件里，删掉那个文件的路径即可，下面是具体文章网址，我自己是通过这个方法解决的。
https://blog.csdn.net/n664500560/article/details/126805410?spm=1001.2101.3001.6650.1&utm_medium=distribute.pc_relevant.none-task-blog-2%7Edefault%7ECTRLIST%7ERate-1-126805410-blog-123376864.235%5Ev40%5Epc_relevant_default_base&depth_1-utm_source=distribute.pc_relevant.none-task-blog-2%7Edefault%7ECTRLIST%7ERate-1-126805410-blog-123376864.235%5Ev40%5Epc_relevant_default_base
